src/interfaces/chain.cpp

```diff
@@ -358,7 +352,16 @@ class ChainImpl : public Chain
     {
         return MakeUnique<NotificationsHandlerImpl>(*this, notifications);
     }
-    void waitForNotifications() override { SyncWithValidationInterfaceQueue(); }
+    void waitForNotificationsIfNewBlocksConnected(const uint256& hash) override
```

Comment:
nit: please rename the argument `old_tip` to be consistent with the function declaration.